src/main/java/Storages/Implementations/RecipeStorageImpl.java

```diff
@@ -49,6 +49,33 @@ public String type() {
 		return found;
 	}
 
+	/**
+	 * Find all the recipes that do not contain the tags given
+	 * This is used because the tags are positive so we want to find recipes
+	 * that do not contain those tags
+	 * @param tags
+	 * @return Collection of recipes containing all the recipes that do not contain the tags
+	 */
+	@Override
+	public Collection<Recipe> findByOppositeTags(Collection<Tag> tags) {
+		Collection<Recipe> found = new ArrayList<>();
+
+		for (Recipe recipe : this.recipes()) {
+			boolean addRecipe = true;
+			for (Tag tag : tags) {
+				// if any of the recipe's tags are one of the tags given, make it false
+				if (recipe.tags().contains(tag)) {
+					addRecipe = false;
+				}
+			}
+			if (addRecipe) {
+				found.add(recipe);
+			}
```

Comment:
This can be simplified alot. The Java Collections interface includes `Collections.disjoint(A,B)` which will tell you if the sets contain any common items.
```
		for (Recipe recipe : this.recipes()) {
			if (Collections.disjoint(recipe.tags(), tags)) {
				found.add(recipe);
			}
		}
```

Also, `findByOppositeTags` is a bad name. There is nothing here that finds "opposites" here. It should be "containsNonOf" perhaps (or anything else you think of, but it should flow when read "in english"). I think it's just that this whole "find" naming scheme is a little weird, might be a good thing to refactor later. Change the doc comment as well, "tags are positive" makes no sense.